File: SIGMA_SIM/scratch/parse_pdf.js
const fs = require('fs');

const data = fs.readFileSync('../../1_TSI/UTIC_CrossRoadInfoService.pdf', 'utf8');
const urls = data.match(/http:\/\/tsihub.utic.go.kr[a-zA-Z0-9\/=?&_]+/g) || [];
console.log([...new Set(urls)]);

File: SIGMA_SIM/scratch/parse_pdf.py
import PyPDF2

try:
    with open('../../1_TSI/UTIC_CrossRoadInfoService.pdf', 'rb') as file:
        reader = PyPDF2.PdfReader(file)
        text = ''
        for page in reader.pages:
            text += page.extract_text()
        print(text)
except Exception as e:
    print("Error:", e)
